### ISSUE TITLE
Why isn't the loss calculated only with b_i=0 values of the Hints.

### ISSUE THREAD

**Issue by javiersgjavi** [jsyoon0823/GAIN#36]
Hello, 

First of all, congratulations four your fantastic paper. I have been reading it and working with your repository; however, I have a doubt and would appreciate it if you could answer it.

I know a couple of changes exist between the implementation described in the original paper and the one in this repository. Also, I have checked some closed issues as #2, where the new implementation of the Hint matrix is described. 

In your original article, when you are talking about how the algorithm works in section 5, it can be seen that the discriminator loss is only calculated with the b_i = 0 of each sample, that is, the positions where there isn't a hint. Also, in the same paragraph, it can be seen that if you train with all the values, the discriminator will overfit to the hint matrix.

Despite this, when I check your code, I have the impression that you calculate D_loss and G_loss with all the values of the hint matrix (b_i=0 and b_i=1) in [the lines 136-139](https://github.com/jsyoon0823/GAIN/blob/ed53e6d0be14a8d4ce35eff46449d4047bcb483e/gain.py#L136-L139). 

I want to ask you if this change is due to the difference in the definition of the Hint matrix and why doesn't this new way of calculating the loss ends in the discriminator overfitting to the hint matrix.

Thank you very much for your attention!

**Comment by jsyoon0823** [comment]
Thanks for your interest in our work.
First, yes. you are correct. We calculate the loss for the components with hint and without hint.
There are pros and cons to do this.
- Cons: As you said, it may overfit to the hint.
- Pros: It helps discriminator to utilize the hint more explicitly.

For the cons, we think that only with hint, the discriminator cannot solve the task perfectly; thus, we think it may not only just overfit on the hint.

Therefore, we focus more on pros and use this as  the official implementation.

Thanks!

**Comment by javiersgjavi** [comment]
Thank you very much for your answer! It has been helpful.
